smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -188,6 +188,36 @@ contract MixinTransfer is
     return fee.mul(transferFeeBasisPoints) / BASIS_POINTS_DEN;
   }
 
+  /**
+  * @notice Modify the expirationTimestamp of a key
+  * by a given amount.
+  * @param _owner The owner of the key to modify
+  * @param _deltaT The amount of time in seconds by which
+  * to modify the keys expirationTimestamp
+  * @param _addTime Choose whether to increase or decrease
+  * expirationTimestamp (false == decrease, true == increase)
+  * @dev Throws if owner does not have a valid key.
+  */
+  function _timeMachine(
+    address _owner,
+    uint256 _deltaT,
+    bool _addTime
+  ) internal
+  {
+    Key storage key = keyByOwner[_owner];
+    uint formerTimestamp = key.expirationTimestamp;
+    bool validKey = getHasValidKey(_owner);
+    if(_addTime) {
+      if(validKey) {
+        key.expirationTimestamp = formerTimestamp.add(_deltaT);
```

Comment:
What if the user has never owned a key previously?  This would not add them to the owners array, for e.g..  Nor issue a transfer event.  

We could maybe add the modifier `ownsOrHasOwnedKey` and then higher level branch to key creation when needed.  Or add key creation logic here.

Reply:
Hmm just a thought: maybe this function could accept a `Key storage _key` instead of the _owner, clarifying that the caller is responsible for ensuring the key exists.

Reply:
Yep, This last suggestion seems like a good fit. I was thinking that the caller should be responsible for ensuring the key exists, but this would make that explicit.

Reply:
Hmm. What about pasing `uint _tokenId` as a proxy for the actual key? This is what we do elsewhere rather than passing actual keys around, and may prove to be simpler (especially for testing) while still making it clear that the key needs to already exist.

Reply:
Seems fine